### PR TITLE
added council page index and show test cases to request tests

### PR DESCRIPTION
**Pull request by sairamya93**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Comment by sairamya93** [comment]
What
LLM’s (Local Links Manager) test suite is RSpec but uses older Controller/Feature style specs instead of the newer Request/System specs. Update to use newer test styles

Why
Standardise code for easier development

Trello card : https://trello.com/c/3BOqpQOH/346-update-test-suite-in-local-links-manager

File: spec/requests/broken_links_page_spec.rb [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+    it "GET edit handles URL passed in via flash" do
+      get "/local_authorities/north-midlands/services/aardvark-wardens/reporting/edit"
+      expect(response).to have_http_status(:ok)
+
+      flash_hash = ActionDispatch::Flash::FlashHash.new
+      flash_hash[:link_url] = "https://www.example.com"
+      session["flash"] = flash_hash.to_session_value
+
+      get "/local_authorities/north-midlands/services/aardvark-wardens/reporting/edit"
+      expect(response).to have_http_status(:ok)
```
Comment (georges1996) [inline spec/requests/broken_links_page_spec.rb:24]: How come we test this on line 16-17 and 23-24?

File: spec/requests/council_page_spec.rb [hunk 2]
```diff
@@ -7,6 +7,26 @@
   it_behaves_like "redirects non-GDS Editors to services page", "/local_authorities/north-midlands/download_links_form"
   it_behaves_like "redirects non-GDS Editors to services page", "/local_authorities/north-midlands/upload_links_form"
 
+  describe "GET #index" do
+    context "when there is sufficient data" do
+      it "returns http succcess" do
+        login_as_gds_editor
+
+        get "/local_authorities"
+        expect(response).to have_http_status(200)
```
Comment (georges1996) [inline spec/requests/council_page_spec.rb]: Is there a reason why we use `expect(response).to have_http_status(:ok)` in other tests but here we use `200`?

File: spec/requests/council_page_spec.rb [hunk 3]
```diff
@@ -31,14 +51,25 @@
     end
   end
 
+  describe "GET bad_homepage_url_and_status_csv" do
+    it "retrieves HTTP success" do
+      login_as_gds_editor
+
+      get "/bad_homepage_url_status.csv"
+      expect(response).to have_http_status(200)
```
Comment (georges1996) [inline spec/requests/council_page_spec.rb]: Same as previous comment https://github.com/alphagov/local-links-manager/pull/1413/files#r1789985844

File: spec/requests/broken_links_page_spec.rb (hunk 1 above)
Comment (sairamya93) [reply spec/requests/broken_links_page_spec.rb:24]: This involves testing 2 controller test cases in 1 test case - **GET edit** and **handling a URL passed in via flash** to make sure flash message is handled

File: spec/requests/council_page_spec.rb (hunk 2 above)
Comment (sairamya93) [reply spec/requests/council_page_spec.rb]: I have updated the tests with  `expect(response).to have_http_status(:ok)` to be consistent with the other test cases

**Review by georges1996 — APPROVED** [review]

**Pull request merged**